backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/RabbitMQConfiguration.java

```diff
@@ -18,18 +18,278 @@
  ****************************************************************/
 package org.apache.james.backends.rabbitmq;
 
+import java.io.File;
 import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.stream.Stream;
 
 import org.apache.commons.configuration2.Configuration;
 
 import com.google.common.base.Preconditions;
 import com.google.common.base.Strings;
 
+import static org.apache.james.backends.rabbitmq.RabbitMQConfiguration.SSLConfiguration.*;
+import static org.apache.james.backends.rabbitmq.RabbitMQConfiguration.SSLConfiguration.SSLValidationStrategy.*;
+
 public class RabbitMQConfiguration {
 
+    public static class SSLConfiguration {
+
+        public enum SSLValidationStrategy {
+            DEFAULT,
+            IGNORE,
+            OVERRIDE;
+
+            static SSLValidationStrategy from(String rawValue) {
+                Preconditions.checkNotNull(rawValue);
+
+                return Stream.of(values())
+                        .filter(strategy -> strategy.name().equalsIgnoreCase(rawValue))
+                        .findAny()
+                        .orElseThrow(() -> new IllegalArgumentException(String.format("invalid strategy '%s'", rawValue)));
+
+            }
+        }
+
+        public enum HostNameVerifier {
+            DEFAULT,
```

Comment:
DEFAUT does not describe a behaviour. Can we find terms to express what this does?

Reply:
SUBJECT_ALTERNATIVE_NAME_VERIFIER ?